tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/GraphQL/HotChocolate/HotChocolateCommon.cs

```diff
@@ -0,0 +1,163 @@
+                // integration disabled, don't create a scope, skip this trace
+                return null;
+            }
+
+            Scope scope = null;
+            try
+            {
+                var queryOperationName = request.OperationName;
+                var source = request.Query?.ToString();
+                var operationType = "Uncompleted";
+                scope = CreateScopeFromExecuteAsync(tracer, IntegrationId, new HotChocolateTags(), ServiceName, queryOperationName, source, operationType);
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, "Error creating or populating scope.");
+            }
+
+            return scope;
+        }
+
+        internal static Scope UpdateScopeFromExecuteAsync(Tracer tracer, IOperationContext context)
+        {
+            if (!tracer.Settings.IsIntegrationEnabled(IntegrationId))
+            {
+                // integration disabled, don't create a scope, skip this trace
+                return null;
+            }
+
+            var scope = tracer.InternalActiveScope;
+            var span = scope?.Span;
+            if (span == null || span.OperationName != ExecuteOperationName)
+            {
+                // not in a Hotchocolate execution span
+                return null;
+            }
+
+            try
+            {
+                var operation = context.Operation;
+                var operationType = operation.OperationType.ToString();
+                var operationName = span.GetTag(Trace.Tags.GraphQLOperationName);
+                span.ResourceName = $"{operationType} {operationName ?? "operation"}";
+                span.SetTag(Trace.Tags.GraphQLOperationType, operationType);
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, "Error creating or populating scope.");
+            }
+
+            return scope;
+        }
+
+        internal static void RecordExecutionErrorsIfPresent(Span span, string errorType, System.Collections.IEnumerable errors)
+        {
+            var executionErrors = GetList(errors);
+            var errorCount = executionErrors?.Count ?? 0;
+
+            if (errorCount > 0)
+            {
+                RecordExecutionErrors(span, errorType, errorCount, ConstructErrorMessage(executionErrors));
+            }
+        }
+
+        private static string ConstructErrorMessage(List<IError> executionErrors)
+        {
+            if (executionErrors == null)
+            {
+                return string.Empty;
+            }
+
+            var builder = Util.StringBuilderCache.Acquire(Util.StringBuilderCache.MaxBuilderSize);
+
+            try
+            {
+                var tab = "    ";
+                builder.AppendLine("errors: [");
+
+                for (int i = 0; i < executionErrors.Count; i++)
+                {
+                    var executionError = executionErrors[i];
+
+                    builder.AppendLine($"{tab}{{");
```

Comment:
nit: avoid interpolation when using string builders as it creates extra garbage - you should remove it from the other usages in this method, but I'll leave that to you 🙂 
```suggestion
                    builder.Append(tab).AppendLine('{');
```